### DISCUSSION TITLE
workers setting in kubernetes

### DISCUSSION THREAD

**Discussion by SpatialJ** [maplibre/martin#983] [Q&A]
Hi,
i understand the workers setting maps to the actix server workers.
Are there best practices for the setting when runnign on kubernetes? Will rust/actix set the 'correct' value by recognizing the enforced ressource limits of the container like java would? Or should one set 1 worker for every vcpu?

Thanks,
Johannes

**Comment by nyurik** [comment]
I think it might be better to keep all settings at default except one critical one - the number of connections to the postgres (assuming you are using that), as that is not something Martin can easily detect.

There are ways to come up with some reasonable defaults based on the server settings. I wrote some code to get the relevant info https://github.com/nyurik/tilelive-pgquery/blob/1285a2ae8e042d15de40001ce51f598370b5f489/lib/PgQuery.js#L359 - so perhaps some of these settings (or others?) might be useful for a good default value - any help is welcome.

**Reply by SpatialJ** [reply]
Thanks for your reply!
I will be using mbtiles, no postgres. As far as i understand one worker will spawn 1 thread. I cannot see why 8 threads (the default) will be a good choice regardless of the ressourcelimits. But maybe this questions i better discused with the actix  people :-)
